api/controllers/checklist-controler.js

```diff
@@ -1,149 +1,147 @@
 /* eslint-disable node/no-unsupported-features/es-syntax */
-const CheckList = require('../models/checklist-model');
+const { Checklist, validateChecklist } = require('../models/checklist-model');
 
 const createCheckList = async (req, res) => {
 
-    try {
-        const { title, sections_data} = req.body;
-        const {sections_data:[{section_title, items_data}]} = req.body;
-        
-        if (!section_title || !items_data) {
-            return res.status(422).json({ message: 'Please, fill up all fields' });
-        }
-        const newList = new CheckList({
-            title,
-            author: req.userData.id,
-            creation_date: new Date(),
-            sections_data
-        })
+  try {
+    const { error } = validateChecklist(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message })
+    }
 
-        const list = await newList.save()
-        res.status(201).json(list);
+    const { title, sections_data } = req.body;
 
-    } catch (error) {
-        res.status(500).json(error);
-    }
+    const newList = new Checklist({
+      title,
+      author: req.userData.id,
+      creation_date: new Date(),
+      sections_data
+    })
+
+    const list = await newList.save()
+    res.status(201).json(list);
+
+  } catch (error) {
+    res.status(500).json(error);
+  }
 };
 
 const createCheckListItem = async (req, res) => {
 
-    try {
-        const { item_title, description, details, tags, priority } = req.body;
+  try {
+    const { item_title, description, details, tags, priority } = req.body;
 
-        const list = await CheckList.findById(req.params.id);
-        const section = await list.sections_data.find(item => item.id === req.params.sectionId);
-        
-        if (!list) return res.sendStatus(404);
+    const list = await Checklist.findById(req.params.id);
+    const section = await list.sections_data.find(item => item.id === req.params.sectionId);
```

Comment:
sections_data.find is not async method
remove await

Reply:
@Antoniozinchenko Done.